src/Mod/TechDraw/Gui/MDIViewPage.cpp

```diff
@@ -325,19 +354,35 @@ void MDIViewPage::printPdf()
     }
 
     Gui::WaitCursor wc;
+    auto vpp = getViewProviderPage();
+    if (!vpp) {
+        // how did this happen?
+        return;
+    }
+
+    savePageExportState(vpp);
+
     std::string utf8Content = fn.toUtf8().constData();
     if (m_pagePrinter) {
         m_pagePrinter->printPdf(utf8Content);
+        resetPageExportState(vpp);

```

Comment:
Pop state, state variable could be local. It would be also cleaner do a push only in case we are about to pop it.